borrowings/serializers.py

```diff
@@ -20,3 +28,28 @@ class BorrowingCreateSerializer(serializers.ModelSerializer):
     class Meta:
         model = Borrowing
         fields = ["id", "book", "expected_return_date", "actual_return_date"]
+
+    def validate(self, attrs) -> dict:
+        expected_return_date = attrs.get("expected_return_date")
+
+        if expected_return_date is not None:
+            validate_borrowing(
+                borrow_date=datetime.date.today(),
+                expected_return_date=expected_return_date,
+                error_to_raise=serializers.ValidationError
+            )
+
+        return attrs
+
+    def create(self, validated_data) -> Borrowing:
+        with transaction.atomic():
+            book_data = validated_data.pop("book")
+            book = Book.objects.get(id=book_data.id)
+            if book.inventory < 1:
+                raise ValidationError("the following book is not available")
+            borrowing = Borrowing.objects.create(
```

Comment:
You add `transaction.atomic()` too early, I think we can add it before this line on 49 line